internal/slack/client.go

```diff
@@ -109,6 +109,21 @@ func NewClient(userFrontend UserFrontend, stdLogger *logging.Logger, mcpClients
 				if providerSettings.MaxResults > 0 {
 					ragConfig["max_results"] = providerSettings.MaxResults
 				}
+				if providerSettings.ScoreThreshold > 0 {
+					ragConfig["score_threshold"] = providerSettings.ScoreThreshold
+				}
+				if providerSettings.RewriteQuery {
+					ragConfig["rewrite_query"] = providerSettings.RewriteQuery
+				}
+				if providerSettings.VectorStoreNameRegex != "" {
+					ragConfig["vector_store_name_regex"] = providerSettings.VectorStoreNameRegex
+				}
+				if providerSettings.VectorStoreMetadataKey != "" {
```

Comment:
1. Inconsistent Configuration Mapping
  ```
  // In config.go you define:
  VectorStoreMetadataKey   string `json:"vectorStoreMetadataKey,omitempty"`
  VectorStoreMetadataValue string `json:"vectorStoreMetadataValue,omitempty"`
  ```
  // But in client.go you map to different keys:
  ```
  ragConfig["vs_metadata_key"] = providerSettings.VectorStoreMetadataKey
  ragConfig["vs_metadata_value"] = providerSettings.VectorStoreMetadataValue
  ```
  This inconsistency will cause configuration failures. Pick one naming convention and stick to it.

Reply:
Thanks @tommynguyen-vungle I would like to do it in this PE-7114 ticket.
So far we are not use config file directly, the code just convert to the legacy format.
```
		// Use the legacy API for now until we properly update the RAG package
		// Convert structured config to legacy format
		ragConfig := map[string]interface{}{
			"provider": cfg.RAG.Provider,
		}
 ```